pkg/validation/test.go

```diff
@@ -41,12 +45,34 @@ func newContext(fieldRoot string, env api.TestEnvironment, releases sets.String)
 	}
 }
 
-func (c *context) forField(name string) context {
+func (c *Context) forField(name string) Context {
 	ret := *c
 	ret.fieldRoot = c.fieldRoot + name
 	return ret
 }
 
+type Args struct {
+	context   *Context
+	testStage TestStage
+}
+
+func NewValidationArgs(context Context, testStage TestStage) Args {
+	return Args{
+		context:   &context,
+		testStage: testStage,
+	}
+}
+
+func NoOpValidationArgs() Args {
+	return Args{
+		context: nil,
+	}
+}
+
+func (v *Args) ShouldValidate() bool {
```

Comment:
I don't think we do any runtime checking if we should validate elsewhere and just always do it. The assumption is that its not possible to get invalid config merged.

Reply:
There were some instances here, like in `resolver.Validate(stepsByName ReferenceByName, chainsByName ChainByName, workflowsByName WorkflowByName, observersByName ObserverByName)` where it is just doing a "shallow" validation where it just checks some high-level components rather than doing the full step validation. So, I just added this `ShouldValidate` as a convenience method for determining whether or not to perform that full step validation - perhaps I should rename it to something like `ShouldPerformLiteralStepValidation` to be more descriptive?